[Sun Yufei] iP

feifeiraindrops opened the pull request:
# BenBen
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

BenBen frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use :+1:
  
All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

feifeiraindrops committed:
"Renamed the bot and changed greeting msg"

feifeiraindrops committed:
echo

feifeiraindrops committed:
list

feifeiraindrops committed:
mark and unmark

feifeiraindrops committed:
added task class

feifeiraindrops committed:
added different types of tasks

feifeiraindrops committed:
automated testing

feifeiraindrops committed:
Exception handling

feifeiraindrops committed:
added deleting feature, replaces array with arraylist

feifeiraindrops committed:
Read and write from local files

feifeiraindrops committed:
Merge branch 'branch-Level-7'

feifeiraindrops committed:
used date and time

feifeiraindrops committed:
Merge branch 'branch-Level-8'

feifeiraindrops committed:
created new class files

feifeiraindrops committed:
added more oop

feifeiraindrops committed:
Merge branch 'branch-A-Moreoop'

feifeiraindrops committed:
added to package

feifeiraindrops committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

feifeiraindrops committed:
Merge branch 'branch-A-Gradle'

feifeiraindrops committed:
Add unit testing

feifeiraindrops committed:
Add javadoc

feifeiraindrops committed:
Improve code quality

feifeiraindrops committed:
Add find function

feifeiraindrops committed:
Merge branch 'branch-A-JavaDoc'

feifeiraindrops committed:
Merge branch 'branch-A-CodingStandard'

feifeiraindrops committed:
Fix merge conflicts

Bryan-Goh commented on src/main/java/benben/Storage.java:
```diff
@@ -0,0 +1,119 @@
+     * @return the array list of the tasks
+     * @throws BenBenException the ben ben exception
+     */
+    public ArrayList<Task> load() throws BenBenException {
```
I am not sure about the length of this method. Perhaps you can shorten this method?

Bryan-Goh commented on src/main/java/benben/Storage.java:
```diff
@@ -0,0 +1,119 @@
+                        nextTask.mark();
+                        list.add(nextTask);
+                        canRead = true;
+                    } else if (strSplit[1].startsWith("0")) {
```
Perhaps you can avoid deep nesting in this method ?

Bryan-Goh commented on src/main/java/benben/Todo.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @Override
+    public boolean equals(Object task) {
```
I like that this method is short and clear

Bryan-Goh commented on src/main/java/benben/BenBen.java:
```diff
@@ -0,0 +1,290 @@
+     * @param str the string command from the user
+     * @throws BenBenException when  the command from the user is of the wrong format
+     */
+    public void event (String str) throws BenBenException{
```
Perhaps the length of this method could be shortened?

Bryan-Goh reviewed (commented):
Looks good to me. Just some minor nit to fix

feifeiraindrops committed:
Update README.md

feifeiraindrops committed:
Add GUI

feifeiraindrops committed:
Merge remote-tracking branch 'origin/master'

feifeiraindrops committed:
Adjust GUI style

feifeiraindrops committed:
Merge branch 'branch-Level-10'

phiphi-tan commented on src/main/java/benben/BenBenException.java:
```diff
@@ -0,0 +1,22 @@
+package benben;
```
I like how you created a specific Exception class for handling improper user inputs, and the toString() method is very cute

phiphi-tan commented on src/main/java/benben/Deadline.java:
```diff
@@ -0,0 +1,82 @@
+    }
+    @Override
+    public String toString() {
+        return "[D] " + super.toString() + " (by: " + getDdl() + ")";
```
for instances like this (as well as below), would using a String.format method work better for readability?

phiphi-tan commented on src/main/java/benben/Storage.java:
```diff
@@ -0,0 +1,119 @@
+                        nextTask.mark();
+                        list.add(nextTask);
+                        canRead = true;
+                    } else if (strSplit[1].startsWith("0")) {
```
seems to have a few nested conditions (while -> for -> if -> if), maybe it could be a bit more abstracted out for easier readability / logic flow

phiphi-tan commented on src/main/java/benben/BenBen.java:
```diff
@@ -0,0 +1,279 @@
+     * @param str the string command from the user
+     * @throws BenBenException when  the command from the user is of the wrong format
+     */
+    public String todo(String str) throws BenBenException{
```
Just wondering, but should these actions be handled by the BenBen class or some other relevant classes? it may be worthwhile to explore abstracting these out into another class

phiphi-tan reviewed (commented):
Overall a very well-made (and cute) project, really like the personalisation as well as the extensive adherence to code guidelines

feifeiraindrops committed:
Add assertion

feifeiraindrops committed:
Move all parsing to Parser class

Doing parsing and adding the task together in one method breaks the abstraction and makes the method too long.

Moving the parsing from file and command to Parser class to improve code quality.

This maintains the abstraction and keeps the code short and simple.

feifeiraindrops committed:
Merge branch 'branch-A-Assertions'

feifeiraindrops committed:
Fix merge conflict

feifeiraindrops committed:
Improve code quality

feifeiraindrops committed:
Add more assert

feifeiraindrops committed:
Merge pull request #1 from feifeiraindrops/branch-A-CodeQuality

Improve code quality

feifeiraindrops committed:
Merge pull request #2 from feifeiraindrops/branch-A-Assertions

Add more assert

feifeiraindrops committed:
Add sorting feature

feifeiraindrops committed:
Merge branch 'branch-C-Sort'

feifeiraindrops committed:
Update user guide and fix other issues

feifeiraindrops committed:
Improve on UI, user guide, and error handling

feifeiraindrops committed:
More improvement

feifeiraindrops committed:
Fix error handling and display of exit message
